### PR TITLE
Fix CSV field size limit

### PR DESCRIPTION
**Pull request by jednano**
I'm not sure if you want to accept this PR or not, but I think I solved the CSV limit issue, referencing the following documents on the Internet:
- https://docs.python.org/3/library/io.html
- https://stackoverflow.com/questions/31869247/understanding-the-buffering-argument-of-the-io-open-method-in-python-2-7
- https://lethain.com/handling-very-large-csv-and-xml-files-in-python/
- https://stackoverflow.com/questions/15063936/csv-error-field-larger-than-field-limit-131072
- https://stackoverflow.com/questions/2104080/how-to-check-file-size-in-python

Note that the new `plantsdict.json` file is considerably larger @ 1,326,728 lines of JSON vs. the previously-existing 47,990 lines of JSON. I have no idea why that is, unless your current `plantsdict.json` was built with something other than the current `USDAsearch.txt` file.

That said, what do you intend to do with this data (i.e., how do you intend to display it)? The kumu tree, right? It might work out that it knows how to decode the `\uXXXX` values, so no need to save the file as UTF-16 LE. I tried saving as UTF-16 LE, but the file was way too big. In order for it to support so many more characters, it just takes way more disc space to do so.

Bottom line? I think saving the JSON file with the `\uXXXX` values is the way to go, especially considering file size.

### CODE REVIEW

File: lcplants.py [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import csv
+import io
```
Comment (jednano) [inline lcplants.py:2]: I don't know why you weren't importing this before, but my Python 3.6.4 was complaining about the lack of it.

File: lcplants.py [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
 import json
 import pprint as pp
+import sys
```
Comment (jednano) [inline lcplants.py:5]: I should have deleted this line. I was using it with `csv.field_size_limit(sys.maxsize)`, which you might still want to do if it works for you!

File: lcplants.py [hunk 3]
```diff
@@ -32,24 +35,18 @@ def dicts(t): return {k: dicts(t[k]) for k in t}
...
 
-
-
-def readcsv(): #runs, but utf-16 characters come through this one as /u00 code
-      with open('USDAsearch.txt', encoding = 'utf-8', errors = 'ignore') as csvfile:
-          reader = csv.DictReader(csvfile, delimiter=',', quotechar='"')
-          return [x for x in reader]
       
-def ioreadcsv(): #trips "Error: field larger than field limit (131072)"
-      with io.open('USDAsearch.txt', 'r', encoding = 'utf-16-le') as csvfile:
```
Comment (jednano) [inline lcplants.py:43]: Your indentation was weird here for some reason. I didn't think Python would run code like this, but it seems to do it!

File: lcplants.py [hunk 4]
```diff
@@ -32,24 +35,18 @@ def dicts(t): return {k: dicts(t[k]) for k in t}
...
+        reader = csv.DictReader(csvfile, delimiter=',', quotechar='"')
+        return [x for x in reader]
 
 def writejson():
-    with open('plantsdict.json', 'w') as outfile:
+    with io.open('plantsdict.json', 'w') as outfile:
         json.dump(plants, outfile, sort_keys=True, indent=4)
 
 def readjson():
-    with open ('plantsdict.json', 'r', encoding='utf8') as jsonfile:
```
Comment (jednano) [inline lcplants.py:52]: `utf8` should be the default, so no need to specify it.

**Comment by violetmadder** [comment]
I agree about the /u00 values, they're relatively easy to handle. 

The plantdict.json file changes often-- ultimately I'll have a main json file that absorbs all the information I collect from all sorts of different sources, and gets updated a lot.  The filenames will probably change around.
